reproducibility/codalab/reproduce.py

```diff
@@ -0,0 +1,605 @@
+            algorithms = (
+                [key for key in algorithm_defaults.keys()]
+                + CodaLabReproducibility._CIVIL_COMMENTS_ADDITIONAL_ALGORITHMS
+                if dataset == "civilcomments"
+                else algorithm_defaults.keys()
+            )
+            for algorithm in algorithms:
+                dataset_results[algorithm] = []
+
+                # Skip the additional datasets if we don't want to reproduce results for all datasets
+                if (
+                    (algorithm in ["IRM", "deepCORAL"] and dataset == "civilcomments")
+                    or algorithm == "groupDRO"
+                    or (in_distribution and algorithm != "ERM")
+                ):
+                    continue
+
+                if dataset == "poverty":
+                    folds = ["A", "B", "C", "D", "E"]
+                    for fold in folds:
+                        bundle_name = (
+                            f"{dataset}_{algorithm}_fold{fold}"
+                            if not in_distribution
+                            else f"hp_id_{algorithm}_{dataset}_fold{fold}"
+                        )
+                        uuid = self._get_bundle_uuid(bundle_name, worksheet_uuid)
+                        results_dfs = load_results(
+                            f"https://worksheets.codalab.org/rest/bundles/{uuid}/contents/blob",
+                            splits=["val", "test"],
+                            include_in_distribution=True,
+                        )
+                        dataset_results[algorithm].append(results_dfs)
+                else:
+                    # 10 different seeds were used for camelyon17. 3 different seeds for other datasets.
+                    seeds = range(0, 10) if dataset == "camelyon17" else [0, 1, 2]
+                    for seed in seeds:
+                        bundle_name = (
+                            f"{dataset}_{algorithm}_seed{seed}"
+                            if not in_distribution
+                            else f"hp_id_{algorithm}_{dataset}_seed{seed}"
+                        )
+                        uuid = self._get_bundle_uuid(bundle_name, worksheet_uuid)
+                        results_dfs = load_results(
+                            f"https://worksheets.codalab.org/rest/bundles/{uuid}/contents/blob",
+                            splits=["val", "test"],
+                            include_in_distribution=True,
+                        )
+                        dataset_results[algorithm].append(results_dfs)
+            compiled_results = compile_results(
+                dataset, dataset_results, in_distribution
+            )
+
+            subdirectory = "main" if not in_distribution else "id_val"
+            with open(
+                f"reproducibility/codalab/output/{subdirectory}/{dataset}_result.json",
+                "w",
+            ) as f:
+                json.dump(compiled_results, f, indent=4)
+
```

Comment:
I used the following to create runs in CodaLab for the ID vs OOD val experiments.